wasmplugin/plugin.go

```diff
@@ -365,3 +394,51 @@ func logError(error ctypes.MatchedRule) {
 		proxywasm.LogDebug(msg)
 	}
 }
+
+type AddressInfo struct {
+	IP   string
+	port int
+}
+
+// Retrieves adddress properties from the proxy
+// Expected targets are "source" or "destination"
+// Envoy ref: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes#connection-attributes
+func RetrieveAddressInfo(target string) AddressInfo {
+	var targetAddress AddressInfo
+	srcAddressRaw, err := proxywasm.GetProperty([]string{target, "address"})
+	if err != nil {
+		proxywasm.LogWarnf("failed to get %s address: %v", target, err)
+	} else {
+		targetAddress.IP, _, err = net.SplitHostPort(string(srcAddressRaw))
```

Comment:
Shouldn't we use the port from the address for when the request property isn't available?

Reply:
I see some value in it for other proxies. The splitted string with the port value is now saved but converted to int only if GetProperty fails.